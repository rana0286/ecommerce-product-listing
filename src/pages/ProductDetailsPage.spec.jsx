import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import ConnectedComponent, { ProductDetailsPage } from './ProductDetailsPage';
import renderer from 'react-test-renderer';

const product = {
  "size": [
    "XS",
    "S",
    "L",
    "XL"
  ],
  "productName": "Striped shirt",
  "productImage": "product-1.jpg",
  "price": "$18.88",
  "isExclusive": false,
  "isSale": true,
  "index": 0
};

const exStore = {
  getState: () => ({
    filteredProductList: [product],
  }),
  subscribe: x => x,
  dispatch: x => x,
};

describe('src/pages/ProductDetailsPage', () => {
  it('ProductDetailsPage component should render', () => {
    const tree = renderer.
      create(<Provider store={exStore}><ConnectedComponent
        headingLabel="Test"
        onSelect={jest.fn()}
        productSizeOptions={[{ value: 'Test', label: 'Test' }]}
        getProductDetails={jest.fn()}
      /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('filterProductDetails action should be call', () => {
    const mockeFilterAction = jest.fn();
    const component = shallow(<ProductDetailsPage
      headingLabel="Test"
      productSizeOptions={[{ value: 'Test', label: 'Test' }]}
      getProductDetails={jest.fn()}
      filterProductDetails={mockeFilterAction}
      filteredProductList={[product]}
    />);
    component.instance().filterProducts('XL');
    expect(mockeFilterAction).toBeCalledWith('XL');
  });
});
