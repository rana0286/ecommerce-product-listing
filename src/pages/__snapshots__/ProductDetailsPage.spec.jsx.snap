// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/ProductDetailsPage ProductDetailsPage component should render 1`] = `
<div
  className="loader"
>
  Loading...
</div>
`;
